paymentsheet/src/main/java/com/stripe/android/customersheet/StripeCustomerAdapter.kt

```diff
@@ -42,16 +43,32 @@ internal class StripeCustomerAdapter @Inject constructor(
         get() = setupIntentClientSecretProvider != null
 
     override suspend fun retrievePaymentMethods(): CustomerAdapter.Result<List<PaymentMethod>> {
+        paymentMethodTypes?.filter { PaymentMethod.Type.fromCode(it) == null }?.takeIf { it.isNotEmpty() }?.run {
+            val cause = IllegalStateException("Invalid payment method types provided (${joinToString()}).")
+            return CustomerAdapter.Result.failure(cause, null)
+        }
+
+        val supportedPaymentMethodCodes = supportedPaymentMethodTypes.map { it.code }.toSet()
+        val unsupportedPaymentMethods = paymentMethodTypes?.minus(supportedPaymentMethodCodes) ?: emptyList()
+        if (unsupportedPaymentMethods.isNotEmpty()) {
+            val unsupportedCodes = unsupportedPaymentMethods.joinToString()
+            val cause = IllegalStateException("Unsupported payment method types provided ($unsupportedCodes).")
+            return CustomerAdapter.Result.failure(cause, null)
+        }
```

Comment:
Two differences between iOS and Android in the impl:
1. On iOS we have CustomerSheetError with specific types. If i read this correctly, everything is of type IllegalStateException, so there's no specific type
2. On iOS, if a payment method deserializes to unknown, we effectively throw the same error -- unsupported.

Reply:
We don't have error types (and it doesn't look like we expose exceptions of this nature for CustomerSheet or PaymentSheet).

I think I'm ok with both of those differences between platforms, but happy to discuss a change if you feel strongly.